### ISSUE TITLE
[doc] LR scheduler example is terrible

### ISSUE THREAD

**Issue by SsnL** [pytorch/pytorch#86208]
### 📚 The doc issue

Look at the official example from https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
```py
model = [Parameter(torch.randn(2, 2, requires_grad=True))]
optimizer = SGD(model, 0.1)
scheduler = ExponentialLR(optimizer, gamma=0.9)

for epoch in range(20):
    for input, target in dataset:
        optimizer.zero_grad()
        output = model(input)
        loss = loss_fn(output, target)
        loss.backward()
        optimizer.step()
    scheduler.step()
```

I argue that it is terrible to make `model` a list of parameters because (1) that is not a model (a model is param + function/rules) and (2) in practice a model is usually a `nn.Module` and this makes the example very confusing to pytorch beginners.

### Suggest a potential alternative/fix

Can we just use a regular `nn.Module` as other examples of the same page (https://pytorch.org/docs/stable/optim.html#how-to-use-an-optimizer) do?

cc @svekars @holly1238 @vincentqb @jbschlosser @albanD

**Comment by daspartho** [comment]
Hey! I'd like to work on this.
I'll open a PR :)

**Comment by daspartho** [comment]
Made a PR #86629 for the issue, could someone please check it?
Thanks =)

**Comment by SsnL** [comment]
fixed
